src/HexagonOptimize.cpp

```diff
@@ -269,6 +278,47 @@ class OptimizePatterns : public IRMutator {
 private:
     using IRMutator::visit;
 
+    string get_suffix(vector<Expr> exprs) {
```

Comment:
I think it would be better to just hardcode the suffix. The function doesn't actually support generic types, and it's 40 lines of code to generate the suffix that would be hardcoded in at most three places (one currently). You can just make the suffix a parameter to the add_mpy_mpy function below.